### ISSUE TITLE
Issue using XCFramework containing static libs inside another framework

### ISSUE THREAD

**Issue by leemaguire** [CocoaPods/CocoaPods#9938]
# Report

## What did you do?

I consumed my podspec from a specific commit, ran pod install, and built.

## What did you expect to happen?

I expect pods to link the xcframework containing static libraries inside our dynamic framework and then for the build to succeed on a sample app after a pod install.

## What happened instead?

Two separate errors are occurring depending on which platform we are building for:

Device:
`ld: warning: directory not found for option '-L/Users/lee.maguire/Library/Developer/Xcode/DerivedData/PodTest-axpvdiprgbsrszgmoawfmzsjtvwx/Build/Products/Debug-iphoneos/XCFrameworkIntermediates/realm'
ld: library not found for -lrealm
clang: error: linker command failed with exit code 1 (use -v to see invocation)`

Simulator:
`Destination binary is symlinked...
Code Signing /Users/lee.maguire/Library/Developer/Xcode/DerivedData/PodTest-axpvdiprgbsrszgmoawfmzsjtvwx/Build/Products/Debug-maccatalyst/PodTest.app/Contents/Frameworks/RealmSwift.framework with Identity Apple Development: Lee Maguire (HGHMPAPEQA)
/usr/bin/codesign --force --sign 5A3681AA3E2AE8590B33489A6BC332466339FF44  --preserve-metadata=identifier,entitlements '/Users/lee.maguire/Library/Developer/Xcode/DerivedData/PodTest-axpvdiprgbsrszgmoawfmzsjtvwx/Build/Products/Debug-maccatalyst/PodTest.app/Contents/Frameworks/RealmSwift.framework'
/Users/lee.maguire/Desktop/PodTest/Pods/Target Support Files/Pods-PodTest/Pods-PodTest-frameworks.sh: line 42: source: unbound variable
Command PhaseScriptExecution failed with a nonzero exit code`

### Stack

```
   CocoaPods : 1.10.0.beta.1
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15.6 (19G73)
       Xcode : 11.6 (11E708)
         Git : git version 2.24.3 (Apple Git-128)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : cocoapods - git - https://github.com/CocoaPods/Specs.git @ 3c14a46ea23dca2b1954f62e6842823789792bca

               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.4.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'PodTest' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!

  # Pods for PodTest


  pod 'RealmSwift', :git => 'https://github.com/realm/realm-cocoa.git', :commit => 'db4e9c434065a41ff6eca6d3a7bae632250666b6', :branch => 'lm/xcframework_cocoapods', submodules: true
  pod 'Realm', :git => 'https://github.com/realm/realm-cocoa.git', :commit => 'db4e9c434065a41ff6eca6d3a7bae632250666b6', :branch => 'lm/xcframework_cocoapods', submodules: true

end
```

## Project that demonstrates the issue

https://github.com/leemaguire/PodTest
Here is our podspec: https://github.com/realm/realm-cocoa/blob/db4e9c434065a41ff6eca6d3a7bae632250666b6/Realm.podspec#L96

**Comment by leemaguire** [comment]
@dnkoutso any ideas?

**Comment by leemaguire** [comment]
@amorde would you have any ideas on this one?

**Comment by leemaguire** [comment]
I've managed to get the project building by commenting out `install_framework "${PODS_XCFRAMEWORKS_BUILD_DIR}/realm/realm.framework"` which is contained in Pods-AppName-frameworks.sh

`if [[ "$CONFIGURATION" == "Debug" ]]; then
  install_framework "${BUILT_PRODUCTS_DIR}/Realm/Realm.framework"
  install_framework "${BUILT_PRODUCTS_DIR}/RealmSwift/RealmSwift.framework"
  #install_framework "${PODS_XCFRAMEWORKS_BUILD_DIR}/realm/realm.framework"
fi
if [[ "$CONFIGURATION" == "Release" ]]; then
  install_framework "${BUILT_PRODUCTS_DIR}/Realm/Realm.framework"
  install_framework "${BUILT_PRODUCTS_DIR}/RealmSwift/RealmSwift.framework"
  install_framework "${PODS_XCFRAMEWORKS_BUILD_DIR}/realm/realm.framework"
fi
if [ "${COCOAPODS_PARALLEL_CODE_SIGN}" == "true" ]; then
  wait
fi`

**Comment by amorde** [comment]
I think this is the same as #9943 which was fixed in #9945. That PR will be shipped in the next beta

**Comment by leemaguire** [comment]
@amorde @dnkoutso Would you be able to give an ETA as to when 1.10 will be released? Thanks.
